pkg/modprovider/module_component.go

```diff
@@ -224,31 +199,6 @@ func newModuleComponentResource(
 
 		// Make sure child resources can read updated state.
 		stateStore.SetNewState(urn, state)
-
-		var errs []error
-		tfState.VisitResources(func(rp *tfsandbox.ResourceState) {
```

Comment:
The RegisterResource calls now start issuing before `tofu apply` completes, which should be much much sooner.